### PR TITLE
Restore transient multiapps by default

### PR DESCRIPTION
**Pull request by GiudGiud**
The default is not safe at all. It is wrong for simulation with states, which can be from material properties

it is wrong for simulations with simply posptprocessors executed on timestep begin (of the subapp, now working with timestep_end data) and transferred on timestep begin (of the main app)

lots of diffs

### CODE REVIEW

**Comment by GiudGiud** [comment]
Test failures are just coverage except with a real error in SAM.
The patch is just adding "no_restore" to the multiapp block in one test (or regolding)
I wont have access to SAM in France. @dogrady2 would you have availability to put up a patch for the failing test?
It's this one
```
    DIFF models/friction_factor_overlap_coupled.porous_energy FAILED (CSVDIFF)
```

**Comment by travismui** [comment]
> Test failures are just coverage except with a real error in SAM. The patch is just adding "no_restore" to the multiapp block in one test (or regolding) I wont have access to SAM in France. @dogrady2 would you have availability to put up a patch for the failing test? It's this one
> 
> ```
>     DIFF models/friction_factor_overlap_coupled.porous_energy FAILED (CSVDIFF)
> ```

@GiudGiud I can take care of this if it still needs to be done. Can this patch be put in now, or is it a breaking change?

**Comment by GiudGiud** [comment]
For now you can pass the new parameter that will use the legacy behavior (no_restore) and allow_unused

Thank you!

**Comment by travismui** [comment]
> For now you can pass the new parameter that will use the legacy behavior (no_restore) and allow_unused
> 
> Thank you!

No problem, patch is ready in SAM/SAM MR!950, should be in later today or tomorrow.

**Comment by GiudGiud** [comment]
Thanks a lot!
@lindsayad this should be good to go. Cancelled test runs have passed before

File: framework/include/multiapps/MultiApp.h [hunk 1]
```diff
@@ -207,7 +207,7 @@ class MultiApp : public MooseObject,
    * Whether or not this MultiApp should be restored at the beginning of
    * each Picard iteration.
    */
-  virtual bool needsRestoration() { return true; }
+  virtual bool needsRestoration() { return !_no_restore; }
```
Comment (lindsayad) [inline framework/include/multiapps/MultiApp.h:210]: Is this overridden anywhere anymore? If not, can remove the `virtual`

Comment (GiudGiud) [reply framework/include/multiapps/MultiApp.h]: done

**Review by lindsayad — APPROVED** [review]

**Pull request merged**